[Term Entry] PHP Array Functions: array_pad() (#4178)
Codecademy/docs: content/php/concepts/arrays/terms/array-pad/array-pad.md

---
Title: 'array_pad()'
Description: 'Pads an array to a specified length with a value.'
Subjects:
  - 'Computer Science'
  - 'Web Design'
  - 'Web Development'
Tags:
  - 'Arrays'
  - 'Functions'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

> The **`array_pad()`** method pads an array to a specified length with a value. If the specified length is positive, the array is padded to the right with the value until the array's size equals to the specified length. Conversely, if the specified length is negative, the array is padded to the left with the value until the array's size equals the absolute value of the specified length. The array will not be padded if the size of the array is more than the absolute value of the specified length.

Reviewer: ```suggestion
The **`array_pad()`** method pads an array to a specified length with a value. If the specified length is positive, the array is padded to the right with the value until the array's size equals the specified length. Conversely, if the specified length is negative, the array is padded to the left with the value until the array's size equals the absolute value of the specified length. The array will not be padded if the size of the array is more than the absolute value of the specified length.
```

> The **`array_pad()`** method is used to pad an array to a specified length with a value. If the specified length is positve, the array is padded to the right with the value until the array's size equals to the specified length. Conversely, if the specified length is negative, the array is padded to the left with the value until the array's size equals to the absolute value of specified length. The array will not be padded if the size of the array is more than the absolute value of specified length.

Reviewer 2: ```suggestion
The **`array_pad()`** method pads an array to a specified length with a value. If the specified length is positive, the array is padded to the right with the value until the array's size equals to the specified length. Conversely, if the specified length is negative, the array is padded to the left with the value until the array's size equals the absolute value of the specified length. The array will not be padded if the size of the array is more than the absolute value of the specified length.
```

## Syntax

```pseudo
array_pad($array, $length, $value)
```

> - `$array`: Specifies the input `array`.
> - `$length`: The new size of `array`.
> - `$value`: The value to be padded to `array` if its size is less than absolute value of `$length`.

Reviewer 2: ```suggestion
- `$array`: Specifies the input `array`.
- `$length`: The new size of `array`.
- `$value`: The value to be padded to `array` if its size is less than the absolute value of `$length`.
```

> The `array_pad()` function returns the padded array according to the specified length with the intended value.

Reviewer 2: ```suggestion
>> **Note:** The `array_pad()` function returns the padded array according to the specified length with the intended value.
```
Lets keep this as Note

## Example

The following example uses the `array_pad()` function to pad the array `$numbers` to different lengths and values:

```php
<?php
  $numbers = array(1, 2, 3);
  $paddedArray = array_pad($numbers, 5, "hello");
  $paddedArray2 = array_pad($numbers, -5, "you");
  $paddedArray3 = array_pad($numbers, 2, "nope");
  print_r($paddedArray);
  print_r($paddedArray2);
  print_r($paddedArray3);
?>
```

> The example will result in the following output:

Reviewer 2: ```suggestion
The example above will result in the following output:
```
